resources/lib/functions.py

```diff
@@ -323,15 +323,18 @@ def getContent(url, params):
     if media_type.startswith("movie"):
         viewType = "Movies"
         xbmcplugin.setContent(pluginhandle, 'movies')
-    elif media_type == "music": # this is albums
+    elif media_type == "music":
+        viewType = "Artists"
+        xbmcplugin.setContent(pluginhandle, 'artists')
+    elif media_type == "musicartist":
         viewType = "Albums"
         xbmcplugin.setContent(pluginhandle, 'albums')
-    elif media_type == "music" or media_type == "audio" or media_type == "musicalbum":
+    elif media_type == "audio" or media_type == "musicalbum":
         viewType = "Music"
         xbmcplugin.setContent(pluginhandle, 'songs')
     elif media_type.startswith("boxset"):
         viewType = "BoxSets"
-        xbmcplugin.setContent(pluginhandle, 'movies')
+        xbmcplugin.setContent(pluginhandle, 'sets')
```

Comment:
I dont think this is correct.
The content is actually movies, the source is a set but the actual content is movies

Reply:
It is. Kodi also uses "Containter.Content(sets)" for the boxsets node.

Reply:
but then they show up as type 'video' in the default skin instead of movies
how is a list of movies anything other than a list of movies
sets would be a list of boxsets

Reply:
But it's the default of "videodb://movies/sets"

See
https://i.imgur.com/YyB5JkN.png?1

Reply:
yes that is a list of sets not the content of a set
that is not what you are setting here though, you are setting content on the list of items in the set

Reply:
Ah, now I get you. It sets "sets" for the sub-folder of a boxset, too.  Both folder paths use ?media_type=boxsets.  
Any idea?

Reply:
basically that code is saying the parent was a boxset so the content is moves
I think what you needed added is "boxsets"

i.e.
if the parent is "boxsets" the content is sets

Reply:
I think this works
    elif media_type.startswith("boxsets"):
        viewType = "Movies"
        xbmcplugin.setContent(pluginhandle, 'sets')

Reply:
Yep.

Reply:
Your suggestion does not work. It still uses "sets" for both nodes. I doubled checked the elif position and if I mixed the values boxset/boxsets.

Reply:
both nodes? which nodes?

Reply:
this should work

Reply:
Addons -> EmbyCon -> Collections (has ?media_type=boxsets)

and

Addons -> EmbyCon -> Collections -> Movie Collection -> Listing  (has ?media_type=boxset)

Reply:
but that is correct
Addons -> EmbyCon -> Collections = a list  of boxsets
and
Addons -> EmbyCon -> Collections -> Movie Collection = a list of movies in a box set

Reply:
Yes, it is. But Container.Content is set to "movies" for both, or to "sets" for both. Root collections node has to be set to "sets"

Reply:
leave boxset = movies and add boxsets = sets

Reply:
That's what I did. Does not work. 

```
elif media_type.startswith("boxsets"):
        viewType = "Movies"
        xbmcplugin.setContent(pluginhandle, 'movies')
    elif media_type.startswith("boxset"):
        viewType = "BoxSets"
        xbmcplugin.setContent(pluginhandle, 'sets')
```

Reply:
you want it the other way around:

    elif media_type.startswith("boxsets"):
        viewType = "Movies"
        xbmcplugin.setContent(pluginhandle, 'sets')
    elif media_type.startswith("boxset"):
        viewType = "BoxSets"
        xbmcplugin.setContent(pluginhandle, 'movies')

Reply:
Oops, yes. Dunno why my brain stopped working. It works. Thanks.